cirq-core/cirq/ops/dense_pauli_string_test.py

```diff
@@ -171,11 +171,12 @@ def test_mul():
 
     # Mixed types.
     m = cirq.MutableDensePauliString
-    assert m('X') * m('Z') == -1j * f('Y')
-    assert m('X') * f('Z') == -1j * f('Y')
+    assert m('X') * m('Z') == -1j * m('Y')
+    assert m('X') * f('Z') == -1j * m('Y')
     assert isinstance(f('') * f(''), cirq.DensePauliString)
-    assert isinstance(m('') * m(''), cirq.DensePauliString)
-    assert isinstance(m('') * f(''), cirq.DensePauliString)
+    assert isinstance(m('') * m(''), cirq.MutableDensePauliString)
+    assert isinstance(m('') * f(''), cirq.MutableDensePauliString)
+    assert isinstance(f('') * m(''), cirq.DensePauliString)
```

Comment:
Why does `m('') * f('')` return `MutableDensePauliString` but `f('') * m('')` return `DensePauliString`?

Reply:
This is because right now, in `__mul__` method we use `type(self)` as the type of the returned dense pauli string. 

I don't have strong opinions so I'll be happy to change the logic to either:
1. Always return a mutable dense pauli string if any of the two parameters being multiplied is a mutable dense pauli string.
2. Always return a non-mutable dense pauli string if any of the two parameters being multiplied is a non-mutable dense pauli string.
3. What we have now. 

Let me know if you have strong opinions or if there's a standard way we should handle multiplication when multiplying different types.

Reply:
Way-1 makes the most sense to me as well.

Reply:
Way 1 sounds good to me to

Reply:
done